civis/parallel.py

```diff
@@ -730,7 +730,12 @@ def _fetch_result(fut):
                 fut.remote_func_output = exc
             else:
                 fut.result_fetched = True
-                if not fut.cancelled() and not fut.exception():
+                cancelled = fut.cancelled()
+                try:
+                    cancelled |= fut._result.is_cancel_requested
```

Comment:
@mheilman It's a bit caveman-like but this works, your thoughts?

Reply:
seems reasonable. could you add a comment, though?

e.g., "After the user requests cancellation, a script run will stay in the running state with "is_cancel_requested" set to True for a while to allow for cleanup logic. Here, we want to treat runs in that state the same way we treat runs in the cancelled state."

Reply:
Good idea